src/Spectre.libClustering.Tests/PartitionTest.cpp

```diff
@@ -0,0 +1,110 @@
+#define GTEST_LANG_CXX11 1
+
+#include <gtest\gtest.h>
+#include <gtest\gtest-spi.h>
+#include <gsl.h>
+#include <vector>
+#include <algorithm>
+#include "Partition.h"
+
+namespace Spectre::libClustering::Tests
+{
+	class PartitionTest : public ::testing::Test
+	{
+	protected:
+
+		std::vector<unsigned int> testData;
+
+		virtual void SetUp()
+		{
+			testData = { 1, 2, 3, 4, 5, 6, 7, 8 };
+		}
+
+	};
+
+	TEST_F(PartitionTest, NotFailsOnNull)
+	{
+		ASSERT_NO_FATAL_FAILURE({
+			Partition test(nullptr);
+		}) 
+			<< "Creation from null data caused fatal failure.";
+	}
+
+	TEST_F(PartitionTest, SimplifyToNotEmpty)
+	{
+		Partition test(testData);
+		auto result = test.Get();
+		ASSERT_NE(result.size(), 0)
+			<< "Sequence has been simplified to empty partition.";
+	}
+
+	TEST_F(PartitionTest, SimplifySortsLabels)
+	{
+		Partition test(testData);
+		auto result = test.Get();
+		ASSERT_TRUE(std::is_sorted(result.begin(), result.end()))
+			<< "Simplification does not sort the labels.";
```

Comment:
The problem is that the testData is already sorted, so it's hard to test if constructor has sorted the labels.